src/suites/cts/validation.spec.ts

```diff
@@ -0,0 +1,673 @@
+      bindings: [
+        {
+          binding: 0,
+          resource: {
+            buffer: t.storageBuffer,
+            offset: bufferConstraint.offset,
+            size: bufferConstraint.size,
+          },
+        },
+      ],
+      layout: bindGroupLayout,
+    });
+
+    await t.expectUncapturedError();
+  }
+});
+
+g.test('validation/layout is an error', async t => {
+  const errorBindGroupLayout = t.device.createBindGroupLayout({
+    bindings: [
+      {
+        binding: 0,
+        visibility: GPUShaderStage.COMPUTE,
+        type: 'storage-buffer',
+      },
+      {
+        binding: 0,
+        visibility: GPUShaderStage.COMPUTE,
+        type: 'storage-buffer',
+      },
+    ],
+  });
+
+  t.device.createBindGroup({
+    bindings: [
+      {
+        binding: 0,
+        resource: {
+          buffer: t.storageBuffer,
+        },
+      },
+    ],
+    layout: errorBindGroupLayout,
+  });
+
+  await t.expectUncapturedError();
+});
+
+g.test('validation/binding index too high in bind group layout', async t => {
+  const bindingIndexTooHigh = 16;
+
+  const bindGroupLayout = t.device.createBindGroupLayout({
+    bindings: [
+      {
+        binding: bindingIndexTooHigh,
+        visibility: GPUShaderStage.COMPUTE,
+        type: 'storage-buffer',
+      },
+    ],
+  });
+
+  t.device.createBindGroup({
+    bindings: [
+      {
+        binding: 0,
+        resource: {
+          buffer: t.storageBuffer,
+        },
+      },
+    ],
+    layout: bindGroupLayout,
+  });
+
+  await t.expectUncapturedError();
+});
+
+g.test('validation/dynamic set to true is not allowed for sampled texture', async t => {
+  const bindGroupLayout = t.device.createBindGroupLayout({
+    bindings: [
+      {
+        binding: 0,
+        visibility: GPUShaderStage.FRAGMENT,
+        type: 'sampled-texture',
+        dynamic: true,
+      },
+    ],
+  });
+
+  t.device.createBindGroup({
+    bindings: [
+      {
+        binding: 0,
+        resource: t.sampledTexture.createView(),
+      },
+    ],
+    layout: bindGroupLayout,
+  });
+
+  await t.expectUncapturedError();
+});
+
+g.test('validation/dynamic set to true is not allowed for sampler', async t => {
+  const bindGroupLayout = t.device.createBindGroupLayout({
+    bindings: [
+      {
+        binding: 0,
+        visibility: GPUShaderStage.FRAGMENT,
+        type: 'sampler',
+        dynamic: true,
+      },
+    ],
+  });
+
+  t.device.createBindGroup({
+    bindings: [
+      {
+        binding: 0,
+        resource: t.sampler,
+      },
+    ],
+    layout: bindGroupLayout,
+  });
+
+  await t.expectUncapturedError();
+});
+
+g.test(
+  'validation/visibility of bindings in bind group layout is not allowed to be none',
+  async t => {
+    const bindGroupLayout = t.device.createBindGroupLayout({
+      bindings: [
+        {
+          binding: 0,
+          visibility: GPUShaderStage.NONE,
+          type: 'storage-buffer',
+        },
+      ],
+    });
```

Comment:
Shouldn't the bind group layout itself be invalid? I don't think you need to create a bind group to test this.